dubbo-common/src/main/java/org/apache/dubbo/common/utils/UrlUtils.java

```diff
@@ -94,23 +96,28 @@ public static URL parseURL(String address, Map<String, String> defaults) {
         int defaultPort = StringUtils.parseInteger(defaults == null ? null : defaults.get(PORT_KEY));
         String defaultPath = defaults == null ? null : defaults.get(PATH_KEY);
         Map<String, String> defaultParameters = defaults == null ? null : new HashMap<>(defaults);
-        if (defaultParameters != null) {
-            defaultParameters.remove(PROTOCOL_KEY);
-            defaultParameters.remove(USERNAME_KEY);
-            defaultParameters.remove(PASSWORD_KEY);
-            defaultParameters.remove(HOST_KEY);
-            defaultParameters.remove(PORT_KEY);
-            defaultParameters.remove(PATH_KEY);
-        }
         URL u = URL.valueOf(url);
-        boolean changed = false;
         String protocol = u.getProtocol();
         String username = u.getUsername();
         String password = u.getPassword();
         String host = u.getHost();
         int port = u.getPort();
         String path = u.getPath();
+        boolean changed = false;
         Map<String, String> parameters = new HashMap<>(u.getParameters());
+        if (defaultParameters != null) {
+            boolean protocolBank = StringUtils.isBlank(protocol);
+            // If the protocol parameter is equal to "nacos", the username and password will not be removed from the collection ;
+            // If the protocol parameter is not equal to "nacos", the username and password will  be removed from the collection ;
+            if (protocolBank || !protocol.equals("nacos")) {

```

Comment:
Is it reasonable proper make `nacos` impl coupling with `dubbo-common` ?

Maybe we can change the param name for `nacos` ?

Reply:
How about we solve this issue by adding duplicate key-value pair to parameters
https://github.com/apache/dubbo/pull/7239